src/Pawn.cpp

```diff
@@ -0,0 +1,30 @@
+#include "Pawn.h"
+
+Pawn::Pawn(bool isWhite)
+	:Piece(isWhite)
+{
+}
+
+bool Pawn::isCorrectMove(const Location& source, const Location& destination) const
+{
+    if (m_firstMove)
+        if (this->getColor())
+            return (destination.x == source.x + 2 || destination.x == source.x + 1) && destination.y == source.y;

```

Comment:
You can initially set the destination.x .... variables instead of calling them each time